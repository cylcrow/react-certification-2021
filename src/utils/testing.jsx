import React from "react";
import { render } from '@testing-library/react';
import { ThemeProvider } from "styled-components";
import { lightTheme } from '../providers/theme/themes';

export const renderWithTheme = (Component, theme = lightTheme) => {
    return render(
        <ThemeProvider theme={{ theme, switchTheme: jest.fn() }}>
        { Component }
        </ThemeProvider>
    );
}