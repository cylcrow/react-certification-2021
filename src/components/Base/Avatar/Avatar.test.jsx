import React from 'react';
import '../../../utils/testing';
import 'jest-styled-components';
import { render, unmountComponentAtNode } from 'react-dom';
import toJson from 'enzyme-to-json';
import { mount, shallow } from 'enzyme';
import Avatar from './Avatar';

let container;
const build = () => {
  render(<Avatar />, container);
};

beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  return container;
});

describe('Avatar', () => {
  it('renders', () => {
    const wrapper = build(<Avatar src="avatar" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

describe('Avatar styles and props', () => {
  it('applies default styling', () => {
    const EXPECTED_SRC = 'testing';
    const tree = mount(<Avatar src={EXPECTED_SRC} />);
    expect(tree).toHaveStyleRule('height', '40px');
    expect(tree).toHaveStyleRule('width', '40px');
    expect(tree).toHaveStyleRule('margin', '8px');
    expect(tree).toHaveStyleRule('border-radius', '50%');
  });

  it('has all passed props', () => {
    const EXPECTED_SRC = 'avatar_url';
    const shallowed = shallow(<Avatar src={EXPECTED_SRC} />);
    expect(shallowed.props().src).toBe(EXPECTED_SRC);
  });
});
