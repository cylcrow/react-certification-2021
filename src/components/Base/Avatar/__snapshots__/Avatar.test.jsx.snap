// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders 1`] = `undefined`;
